dll/win32/msgina/msgina.c

```diff
@@ -161,6 +161,57 @@ ChooseGinaUI(VOID)
     HeapFree(GetProcessHeap(), 0, SystemStartOptions);
 }
 
+static BOOL
+SafeGetUnicodeString(_In_ const LSA_UNICODE_STRING *pInput,
+    _Out_ PWSTR pszOutput,
+    _In_ SIZE_T cchMax)
+{
+    if (pInput && pszOutput)
+    {
+        PWSTR pszInput = pInput->Buffer;
+        SIZE_T cchInput = pInput->Length / sizeof(WCHAR);
+        if (cchInput < cchMax && (pszInput || !cchInput))
+        {
+            CopyMemory(pszOutput, pszInput, cchInput * sizeof(WCHAR));
+            pszOutput[cchInput] = UNICODE_NULL;
+            return TRUE;
+        }
+    }
+    if (pszOutput && cchMax)
+        *pszOutput = UNICODE_NULL;
+    return FALSE;
```

Comment:
Side-note: In case passwords can only be text passwords:
[Since msgina includes `strsafe.h`](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/msgina/msgina.h;h=b644aa36ff9f512f0f881bac428238f3e85fc7f9;hb=HEAD#l24), this SafeGetUnicodeString function would be equivalent to this:
```c
    HRESULT Status;
    Status = StringCbCopyNExW(pszOutput, cchMax * sizeof(WCHAR),
                              pInput->Buffer, pInput->Length,
                              NULL, NULL,
                              STRSAFE_NO_TRUNCATION | STRSAFE_NULL_ON_FAILURE);
    return (Status == S_OK);
```